tests/decide_gi.py

```diff
@@ -46,48 +34,120 @@ def test_balanced_or_bijected(graph: 'Graph', is_balanced: 'Bool', is_bijected:
-                fail("{0}".format(err))
-                print('')
-print('---------------------------')
-print("Statistics of test is_balanced_or_bijected:")
-print("Amount of graphs tested: " + str(graph_count) + "/" + str(total_tests))
-print("Amount of graphs failed: " + str(error_count))
-print('')
-if error_count > 0 or not total_tests == graph_count:
-    fail('TEST FAILED')
-else:
-    print('TEST PASSED')
-print('')
+def _test_decide_gi(test_name, color_refine_object: "Object"):
+    csv_filepath = create_csv_file(test_name)
+    print('<' + test_name + '> ' + 'Appending to CSV: ' + "file:///" + csv_filepath.replace('\\', '/') + '\nStart...')
+
+    write_csv_line(csv_filepath, ['file', 'j', 'i', 'Pass?', 'Time (s)'])
+
+    # Expected test results
+    total_tests = 22
+
+    files = ['colorref_smallexample_2_49', 'colorref_smallexample_4_7', 'colorref_smallexample_6_15']
+
+    # Dictionary containing expected outcomes.
+    # Sourced from canvas.utwente.nl
+    # Structure:
+    # The dictionary is keyed on filename (str)
+    # This dictionary contains for each key:
+    #   A dictionary keyed on graph no. (int)
+    #   This dictionary contains:
+    #       A dictionary container on graph no. (int)
+    #       This dictionary contains:
+    #           Tuple containing : (is_balanced, is_bijected) ((bool, bool))
+
+    expected_results = dict()
+    expected_results['colorref_smallexample_2_49'] = {0: {1: (True, True)}}
+    expected_results['colorref_smallexample_4_7'] = {0: {1: (False, False), 2: (True, False), 3: (False, False)},
+                                                     1: {2: (False, False), 3: (True, True)},
+                                                     2: {3: (False, False)}
+                                                     }
+    expected_results['colorref_smallexample_6_15'] = {
+        0: {1: (True, True), 2: (False, False), 3: (False, False), 4: (False, False), 5: (False, False)},
+        1: {2: (False, False), 3: (False, False), 4: (False, False), 5: (False, False)},
+        2: {3: (True, True), 4: (False, False), 5: (False, False)},
+        3: {4: (False, False), 5: (False, False)},
+        4: {5: (True, False)}
+        }
+
+    # Testing loop
+    # A disjoint union is coloured using color_refinement(degree_color_initialization(graph))
+    # The method is_balanced_or_bijected is tested against a hardcoded dictionary containing the expected outcomes.
+    #
+    # Currently the following graph files are tested:
+    #     - colorref_smallexample_2_49
+    #     - colorref_smallexample_4_7
+    #     - colorref_smallexample_6_15
+
+    error_count = 0
+    graph_count = 0
+    total_time = 0
+
+    for file in files:
+        filename = 'test_graphs/color_refinement/' + file + '.grl'
+        graphs = load_graph_list(filename)
+
+        for i in range(0, len(graphs)):
+            for j in range(0, i):
+                t0 = time.time()
```

Comment:
Eigenlijk willen we de tijd weten van de is_balanced_or_bijected methode alleen, niet van de color_initialization en color_refinement erbij. Want dat zijn niet de methoden die hier getest worden

Reply:
klopt helemaal! de timer staat nu na de color_refinement.